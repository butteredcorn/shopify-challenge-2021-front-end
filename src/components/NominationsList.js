import { useState, useEffect } from 'react'
import styled from 'styled-components'

import '../styles/NominationsList.css'

const NominationsContainer = styled.div`
    background-color: ${props => props.theme.bg.secondary};
    color: ${props => props.theme.text.tertiary};
    border: 1px solid ${props => props.theme.bg.primary};
`

const NominationsList = ({loading, error, data}) => {

    if (loading) return null;
    if (error) return `${error}.`;

    return (
        <NominationsContainer className="container">
            <div>
                <h3 className="heading">Nominations</h3>
                {data && data.length > 0 && 
                <ul className="nominations">
                    {data.map((nomination) => 
                    <li className="nomination" key={nomination.imdbID}><span>{nomination.Title}</span> <span>({nomination.Year})</span> <span><button>Remove</button></span></li>
                    )}
                </ul>}
                {data && data.length == 0 &&
                <p>No Nominations Yet! Nominate some movies to get started.</p>}
            </div>
        </NominationsContainer>
    )
}

export default NominationsList;