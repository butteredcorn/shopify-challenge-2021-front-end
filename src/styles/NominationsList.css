.container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 1rem;
}

.container > * {
    width: 100%;
}

.heading {
    text-align: left;
}

.nominations {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
}

.nomination { 
    margin-top: 0.5rem;
    margin-bottom: 0.5rem;
}

.nomination > * {
    padding-right: 0.5rem;
}