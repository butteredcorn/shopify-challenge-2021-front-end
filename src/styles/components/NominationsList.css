.nominations-container {
    display: flex;
    flex-direction: column;
    padding: 1rem;
}

.heading {
    text-align: left;
}

.nominations {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
}

.nomination { 
    margin-top: 0.5rem;
    margin-bottom: 0.5rem;
}

.nomination > * {
    padding-right: 0.5rem;
}

@media only screen and (max-width: 600px) {

    .nominations {
        list-style: none;
    }

    .nomination > * {
        padding-right: 0;
    }
}