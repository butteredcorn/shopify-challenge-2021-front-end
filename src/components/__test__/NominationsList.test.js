import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import ThemesProvider from '../../providers/ThemeProvider'
import NominationsList from '../NominationsList';

afterEach(cleanup);

it('renders NominationsList component with "No Nominations Yet! Nominate some movies to get started." if prop "data" is set to empty array.', () => {
  render(<ThemesProvider><NominationsList data={[]} /></ThemesProvider>);
  const message = screen.getByText(/No Nominations Yet! Nominate some movies to get started./i);
  expect(message).toBeInTheDocument();
});

it('renders NominationsList component without "No Nominations Yet! Nominate some movies to get started." if prop "data" is set to an array with two movie objects.', () => {
    render(<BrowserRouter><ThemesProvider><NominationsList data={[{Title: "Shopify Challenge - Front End Developer!", Year: "2021", imdbID: "the_shoppies"}, {Title: "Hello World!", Year: "2021", imdbID: "someID"}]} /></ThemesProvider></BrowserRouter>);
    const nonExistantMessage = screen.queryByText(/No Nominations Yet! Nominate some movies to get started./i);
    expect(nonExistantMessage).toBeNull();
});

it('expect NominationsList to be null if loading is true', () => {
    const {container} = render(<ThemesProvider><NominationsList loading={true} data={[{Title: "Shopify Challenge - Front End Developer!", Year: "2021", imdbID: "the_shoppies"}, {Title: "Hello World!", Year: "2021", imdbID: "someID"}]} /></ThemesProvider>);
    expect(container.firstChild).toBeNull();
});

it('expect NominationsList to render an error if an error is passed in as a prop', () => {
    render(<ThemesProvider><NominationsList error={new Error("Hello World!")} data={[{Title: "Shopify Challenge - Front End Developer!", Year: "2021", imdbID: "the_shoppies"}, {Title: "Hello World!", Year: "2021", imdbID: "someID"}]} /></ThemesProvider>);
    const message = screen.getByText(/Hello World!/i);
    expect(message).toBeInTheDocument();
});

it('matches a snapshot', () => {
  const json = renderer.create(<BrowserRouter><ThemesProvider><NominationsList data={[{Title: "Shopify Challenge - Front End Developer!", Year: "2021", imdbID: "the_shoppies"}, {Title: "Hello World!", Year: "2021", imdbID: "someID"}]}/></ThemesProvider></BrowserRouter>).toJSON();
  expect(json).toMatchSnapshot();
});