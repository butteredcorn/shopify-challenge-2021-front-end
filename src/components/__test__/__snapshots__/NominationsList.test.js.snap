// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<div
  className="sc-gsDJrp dbPJRZ nominations-container"
>
  <h3
    className="sc-dkPtyc iJlIGG heading"
  >
    Nominations
  </h3>
  <ul
    className="nominations"
  >
    <li
      className="nomination"
    >
      <a
        href="/movie"
        onClick={[Function]}
      >
        Shopify Challenge - Front End Developer!
      </a>
       
      <span>
        (
        2021
        )
      </span>
       
      <span>
        <button
          onClick={[Function]}
        >
          Remove
        </button>
      </span>
    </li>
    <li
      className="nomination"
    >
      <a
        href="/movie"
        onClick={[Function]}
      >
        Hello World!
      </a>
       
      <span>
        (
        2021
        )
      </span>
       
      <span>
        <button
          onClick={[Function]}
        >
          Remove
        </button>
      </span>
    </li>
  </ul>
</div>
`;
